src/realm/util/thread.cpp

```diff
@@ -114,6 +133,27 @@ bool Thread::get_name(std::string& name) noexcept
     name_2[max - 1] = '\0';              // Eliminate any risk of buffer overrun in strlen().
     name.assign(name_2, strlen(name_2)); // Throws
     return true;
+#elif defined(_WIN32)
+    // GetThreadDescription was only introduced in Windows 10 1607, so we need to dynamically load it.
+    HRESULT(WINAPI * GETTHREADDESCRIPTION)(HANDLE hThread, PWSTR * threadDescription) = nullptr;
+#if !defined(NTDDI_WIN10_RS1) || NTDDI_VERSION < NTDDI_WIN10_RS1
+    auto proc_address = GetProcAddress(GetModuleHandle(L"Kernel32.dll"), "GetThreadDescription");
+    if (!proc_address) {
+        return false;
+    }
+    GETTHREADDESCRIPTION = reinterpret_cast<decltype(GETTHREADDESCRIPTION)>(proc_address);
+#else
+    GETTHREADDESCRIPTION = &GetThreadDescription;
+#endif
+    PWSTR name_wide;
+    if (SUCCEEDED(GETTHREADDESCRIPTION(GetCurrentThread(), &name_wide))) {
+        size_t length_wide = wcslen(name_wide);
+        name.resize(WideCharToMultiByte(CP_UTF8, 0, name_wide, length_wide, nullptr, 0, nullptr, nullptr));
+        WideCharToMultiByte(CP_UTF8, 0, name_wide, length_wide, name.data(), name.size(), nullptr, nullptr);
+        LocalFree(name_wide);
+        return true;
+    }
+    return false;
```

Comment:
As usual, I suggest flipping the condition and doing an early return, rather than putting the happy path inside the if.